server/eclipse-project/src/main/java/us/freeandfair/corla/model/DoSDashboard.java

```diff
@@ -206,25 +206,27 @@ public boolean removeContestsToAuditForCounty(final County the_county) {
    */
   //@ requires the_contest_to_audit != null;
   public boolean updateContestToAudit(final ContestToAudit the_contest_to_audit) {
-    boolean found = false;
     boolean auditable = true;
     
     // check to see if the contest is in our set
-    final Set<ContestToAudit> contests_to_remove = new HashSet<>();
+    ContestToAudit contest_to_remove = null;
     for (final ContestToAudit c : my_contests_to_audit) {
       if (c.contest().equals(the_contest_to_audit.contest())) {
         auditable = !c.audit().equals(AuditType.NOT_AUDITABLE);
         // flag the entry for removal, only if it is auditable
-        contests_to_remove.add(c);
-        found = true;
+        contest_to_remove = c;
+        break;
       }
     }
-    my_contests_to_audit.removeAll(contests_to_remove);
-    if (auditable && the_contest_to_audit.audit() != AuditType.NONE) {
-      my_contests_to_audit.add(the_contest_to_audit);
+    
+    if (auditable) {
+      my_contests_to_audit.remove(contest_to_remove);
```

Comment:
Suppose `the_contest_to_audit` has audit type `NOT_AUDITABLE`. Do we expect this to end up in `my_contests_to_audit`?

Reply:
Can a contest in `my_contests_to_audit` _ever_ have a contest type of `NOT_AUDITABLE`? If not, should we really be checking `if (!the_contest_to_audit.audit().equals(AuditType.NOT_AUDITABLE))`?

Reply:
Yes, it can have type `NOT_AUDITABLE`; if it does, it was placed there at CVR import time, and can only be removed from the list if the contest itself goes away (by virtue of the CVR import being overwritten by a new one), using the method `removeContestsToAuditForCounty`.

Reply:
Remember that the audit type of the parameter to that method is the _desired_ audit type for the list (so, to change a contest from `COMPARISON` to `HAND_COUNT`, you pass in the same contest in a `ContestToAudit` object with audit type `HAND_COUNT`).

Reply:
@dmzimmerman that makes sense. I'm not sure what the answer is to my very first question: if `the_contest_to_audit` has audit type `NOT_AUDITABLE` (which, like you said, is what we _want_ its audit type to _become_), do we expect it to end up in `my_contests_to_audit`?

Reply:
Yes, it always ends up there at CVR import time.

Reply:
When the CVR import process calls this method with a type of `NOT_AUDITABLE`, the contest does in fact end up in the list.

Reply:
Okay, so suppose a contest `c` is `equal` to `the_contest_to_audit` and exists in `my_contests_to_audit`. Either it is auditable or it is not (i.e. has audit type `NOT_AUDITABLE`).

If it is auditable, then we will remove `c` from `my_contests_to_audit`. Then, as long as `the_` _doesn't_ have audit type `NONE`, it will be added to `my_contests_to_audit`, either of which is an "update".

If `c` is _not_ auditable, then it has audit type `NOT_AUDITABLE`. That means we won't enter the true branch in `if (auditable) { ...`, and so we will neither remove `c` nor add `the_`. That means, in particular, we can never set a `NOT_AUDITABLE` contest to have any other type, which was the goal. It also means we can't use this method to update any other data which `equals` is insensitive to, so I guess we just need to be sure that is fine.

Reply:
Note that the _equals()_ is comparing the _contests_ in the `ContestToAudit` objects, not the `ContestToAudit` objects themselves. So I don't think there is any other data to consider.

Reply:
@dmzimmerman right! Okay, so, like I said, that means if we have a `contest` that is ever set to `NOT_AUDITABLE`, we can't, for example, ever update the `reason` field of the associated `ContestToAudit` in `my_contests_to_audit`, but I'm pretty sure we don't care about that at all.